### ISSUE TITLE
Directory Conflict Resolving

### ISSUE THREAD

**Issue by brillout** [jrson83/create-vite-plugin-ssr#1]
How about this:
 1. By default it creates `vite-app/`. The user cannot change that (with the exception explained in 3.).
 1. If `vite-app` already exists, a yellow warning is shown:
     > `vite-app/` already exists and will be overwritten.
     
     The only way for the user to avoid the dir to be overwritten is to exit the CLI. In other words, a single `<enter>` press is still enough to confirm everything.
 1. In the warning we can inform the user about a shortcut:
      > Press `<r>` to rename the created directory `vite-app/`.

If the warning is not shown there is no mention that the name of the directory is `vite-app/`, and we don't inform the user about the `<r>` shortcut. It just creates it without prompting the user. The idea here is that users can `mv` the directory if they want to rename the directory. (Note we don't define any `package.json#name`; it's useless for apps.)

Let me know if you have any thoughts or objections.

**Comment by jrson83** [comment]
Yeah, I was thinking almost the same, as you said there should be only a default dir name.

If the default project directory will be `./vite-app` and the directory already exists **and** has files in it, we must pause the create-app process and we then should inform the user and provide a way to solve the conflict.

Since you didn't like `ink-selects` that much, I think the warning message would be good from the current generator in combination:

`⚠ Target directory vite-app/ is not empty.`
`Press <r> to rename the directory, <return> to remove existing files and continue.`

This is all in all a good idea, I will implement it.

**Comment by brillout** [comment]
👍 

> we must pause the [...] process

I think showing the warning right away at the first screen will be enough.

So we keep one screen and one step.

**Comment by jrson83** [comment]
> I think showing the warning right away at the first screen will be enough.

Ah Okay, I didn't get that. So you mean, when the user executes the CLI, it auto checks with `fs.existsSync` if the directory `vite-app` already exists inside the current working directory? And if so then shows the warning before the user can interact with the selectOptions?

**Comment by brillout** [comment]
> And if so then shows the warning before the user can interact with the selectOptions?

How about we show the warning below the selectOptions table? So that everything stays single-step.

> I think target dir would for sure be better

Yes.

**Comment by jrson83** [comment]
Okay, I'll post a preview if its ready.

**Comment by jrson83** [comment]
First scenario:
> Directory `vite-app` exists and has files in it. The user presses `<r>` and renames the `targetDir`.

![2022-04-16_20h42_17](https://user-images.githubusercontent.com/45284565/163687831-805e879a-2e95-4918-a9ba-5f9c480f099f.gif)

Second scenario:
> Directory `vite-app` exists and has files in it. User presses `<return>`.

![2022-04-16_20h43_47](https://user-images.githubusercontent.com/45284565/163687892-1d1aafa5-9707-453f-af72-3d5f69fdb3f8.gif)

Side note: The programming effort is much more complex, if we do it like this.

**Comment by brillout** [comment]
Neat 👍. That's what I imagined.

> I would still prefer to have the default option to set a directory.

Why? What's wrong with simply doing `mv vite-app my-new-project`?

**Comment by jrson83** [comment]
> Neat 👍. That's what I imagined.

Nice, I'm glad you like it 👍 

> Why? What's wrong with simply doing mv vite-app my-new-project?

Well, it's Okay, I guess it might be unfamiliar to some users, since most `create-apps` offer that feature.
But if we just indicate at the installer Docs, that the CLI does not accept `args` I'll be fine. :satisfied:

I'll optimize the code and update the repo later today.
